core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigGenerationBuildStep.java

```diff
@@ -631,6 +631,19 @@ private static void generateConfigBuilder(
             }
 
             for (ConfigClassWithPrefix mapping : mappings) {
+                boolean isPresentAtDeployment = isClassAvailable(deploymentCl,
+                        mapping.getKlass().getName());
+
+                // necessary because core/runtime configurations does no exist on QuarkusClassLoader.
+                if (!mapping.getKlass().getName().startsWith("io.quarkus.runtime.")) {
```

Comment:
Wouldn't it be easier to check, if mapping is `RUNTIME` and `QuarkusClassLoader.isClassPresentAtRuntime`? We may want to include `BUILD_AND_RUNTIME_FIXED` also.